### PR TITLE
Notifications: Don't fail on data inconsistency when retrieving the s…

### PR DESCRIPTION
**Pull request by thet**
…ession url.

### CODE REVIEW

File: src/euphorie/client/notifications/base.py [hunk 1]
```diff
@@ -43,12 +47,24 @@ def recipient(self):
...
+            session_url = ""
+            try:
+                session_url = session.absolute_url()
+            except ValueError:
+                logger.warn(
+                    "Could not get session URL for session %s and "
+                    "zodb_path %s (session id: %s). There might be a data "
+                    "inconsistency. Not including this session in the "
+                    "notification mailing.",
+                    session.title,
```
Comment (ale-rt) [inline src/euphorie/client/notifications/base.py:62]: Please do not use `%s`  when substituting what could be a multiline text, use `%r` we were already been asked to do something similar.

**Review by ale-rt — CHANGES_REQUESTED** [review]

File: src/euphorie/client/notifications/base.py [hunk 2]
```diff
@@ -43,12 +47,24 @@ def recipient(self):
...
-            ]
-        )
+
+        session_links = ""
+        for session in self.sessions:
+            session_url = ""
+            try:
+                session_url = session.absolute_url()
+            except ValueError:
+                logger.warn(
```
Comment (ale-rt) [inline src/euphorie/client/notifications/base.py:57]: Isn't it better to have a:
```suggestion
                logger.exception(
```
How this inconsistency can be fixed?

File: src/euphorie/client/notifications/base.py (hunk 1 above)
Comment (thet) [reply src/euphorie/client/notifications/base.py:62]: Done. would the title for example be multiline? and how would `__repr__` over `__str__` make a difference?
However, I have changed that.

Comment (thet) [reply src/euphorie/client/notifications/base.py:62]: okay, there's a difference in escaping:

```
>>> "ab %s c" % """okay
... nay"""
'ab okay\nnay c'

>>> "ab %r c" % """okay
... nay"""
"ab 'okay\\nnay' c"
```

File: src/euphorie/client/notifications/base.py (hunk 2 above)
Comment (thet) [reply src/euphorie/client/notifications/base.py:57]: done.

**Comment by ale-rt** [comment]
I still do get why all of a sudden we have this unconsistency...
Is it a possibility that we have session that do not resolve a proper path?

**Comment by thet** [comment]
@reinhardt I now changed handling of missing titles from:

Screenshot from 2023-12-01 16-06-44

to:

Screenshot from 2023-12-01 16-12-03

**Pull request merged**